cmd/oras/root/push.go

```diff
@@ -224,7 +224,7 @@ func runPush(cmd *cobra.Command, opts *pushOptions) error {
 	copy := func(root ocispec.Descriptor) error {
 		// add both pull and push scope hints for dst repository
 		// to save potential push-scope token requests during copy
-		ctx = registryutil.WithScopeHint(ctx, dst, auth.ActionPull, auth.ActionPush)
+		ctx = registryutil.WithScopeHint(ctx, originalDst, auth.ActionPull, auth.ActionPush)
```

Comment:
That is going to break the progress bar. I need to investigate more what you are trying to fix because this should work for you.